pkg/cmd/backuprestore/backupnoconfig.go

```diff
@@ -0,0 +1,165 @@
+package backuprestore
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"k8s.io/client-go/rest"
+	"slices"
+
+	backupv1alpha1 "github.com/openshift/api/config/v1alpha1"
+	backupv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1alpha1"
+	prunebackups "github.com/openshift/cluster-etcd-operator/pkg/cmd/prune-backups"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+
+	"github.com/adhocore/gronx/pkg/tasker"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+type backupNoConfig struct {
+	schedule  string
+	timeZone  string
+	retention backupv1alpha1.RetentionPolicy
+	scheduler *tasker.Tasker
+	backupOptions
+}
+
+func NewBackupNoConfigCommand(errOut io.Writer) *cobra.Command {
+	backupNoConf := &backupNoConfig{
+		backupOptions: backupOptions{errOut: errOut},
+	}
+
+	cmd := &cobra.Command{
+		Use:   "backup-server",
+		Short: "Backs up a snapshot of etcd database and static pod resources without config",
+		Run: func(cmd *cobra.Command, args []string) {
+			must := func(fn func() error) {
+				if err := fn(); err != nil {
+					if cmd.HasParent() {
+						klog.Fatal(err)
+					}
+					fmt.Fprint(backupNoConf.errOut, err.Error())
+				}
+			}
+
+			must(backupNoConf.Validate)
+			must(backupNoConf.Run)
+		},
+	}
+	backupNoConf.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func (b *backupNoConfig) AddFlags(fs *pflag.FlagSet) {
+	b.backupOptions.AddFlags(fs)
+}
+
+func (b *backupNoConfig) Validate() error {
+	return b.backupOptions.Validate()
+}
+
+func (b *backupNoConfig) Run() error {
```

Comment:
You don't need to use this scheduler here, but rather you want to just write a simple controller, similar in concept to https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/cronjob/cronjob_controllerv2.go. 

The main bits are:
1. Use a DelayingQueue from https://github.com/kubernetes/client-go/blob/master/util/workqueue/delaying_queue.go (`TypedNewDelayingQueue`, for example, or others if you need more control). Additionally you'll register an informer on that Backup CR of yours, which will be responsible to trigger worker any time a change to a CR will happen. 
2. After that you'll create a `Run` method very similar to the cronjob controller, there you'll start the queue and your worker.
3. Inside the worker you'll either schedule the backup to run or you'll queue (using `AddAfter` method) the next execution. To calculate the actual next run you can use `github.com/robfig/cron` library which helps parsing the cron spec (`cron.ParseStandard`) .